Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -5605,6 +6640,21 @@ class BBQJIT {
             m_jit.loadPtr(srcLocation.asAddress(), m_dataScratchGPR);
             m_jit.storePtr(m_dataScratchGPR, dst.asAddress());
             break;
+        case TypeKind::V128: {
+            // We do two scalar load/store pairs instead of one vector load/store in order to avoid
```

Comment:
Why can't we just clobber FPR scratch? I can't find where we need fprScratch to be live across a move here, where does that happen? Also, why doesn't this switch support doubles?

Reply:
Seems like it's fine for us to clobber FPR at the moment - there's a few locations where we can get subtle bugs if we remove `m_dataScratchGPR` in general though. I've switched it to use `m_scratchFPR`, and it doesn't seem to be a regression.